streaming/src/test/scala/org/apache/spark/streaming/scheduler/ReceiverTrackerSuite.scala

```diff
@@ -81,17 +89,25 @@ private class RateLimitInputDStream(@transient ssc_ : StreamingContext)
 }
 
 /**
- * A Receiver as an object so we can read its rate limit.
+ * A Receiver as an object so we can read its rate limit. Make sure to call `reset()` when
+ * reusing this receiver, otherwise a non-null `executor_` field will prevent it from being
+ * serialized when receivers are installed on executors.
  *
  * @note It's necessary to be a top-level object, or else serialization would create another
  *       one on the executor side and we won't be able to read its rate limit.
  */
-private object SingletonDummyReceiver extends DummyReceiver(0)
+private[streaming] object SingletonDummyReceiver extends DummyReceiver(0) {
+
+  /** Reset the object to be usable in another test. */
+  def reset(): Unit = {
+    executor_ = null
+  }
+}
 
 /**
  * Dummy receiver implementation
  */
-private class DummyReceiver(receiverId: Int, host: Option[String] = None)
+private[streaming] class DummyReceiver(receiverId: Int, host: Option[String] = None)
```

Comment:
Since this receiver and associated InputDStream is used in multiple testsuites, mind renaming this to a more contextually relevant name than `Dummy`? How about `RateTestReceiver` and `RateTestInputDStream`?